src/business-logic/services/execute-action-service.ts

```diff
@@ -210,6 +210,6 @@ export class ExecuteActionService implements ActionService {
     }
 
     const piece: Piece = this.createPieceFromAction(action as PieceAction)
-    await this.rundownService.replacePiece(rundownId, pieceFromRundown, piece)
+    await this.rundownService.replacePieceOnAirOnOnNextPart(rundownId, pieceFromRundown, piece)
```

Comment:
A little typo: `OnOn`. There can also be inserted an `In` between `Piece` and `OnAir`:
`replacePieceInOnAirOrNextPart`
The same applies to the other renamings.